diff --git a/fdbsyncd/fdbsync.cpp b/fdbsyncd/fdbsync.cpp
index 45915f40e..9a88e557b 100644
--- a/fdbsyncd/fdbsync.cpp
+++ b/fdbsyncd/fdbsync.cpp
@@ -106,7 +106,7 @@ void FdbSync::processStateFdb()
         auto mac_address = key.substr(delimiter+1);
 
         info.vid = vlan_name;
-        memcpy(info.mac, mac_address.c_str(),mac_address.length());
+        info.mac = mac_address;
 
         if(op == "SET")
         {
@@ -126,7 +126,7 @@ void FdbSync::processStateFdb()
 
             if(fvField(i) == "port")
             {
-                memcpy(info.port_name, fvValue(i).c_str(), fvValue(i).length());
+                info.port_name = fvValue(i);
             }
 
             if(fvField(i) == "type")
diff --git a/fdbsyncd/fdbsync.h b/fdbsyncd/fdbsync.h
index 1d79af95a..c8248ffef 100644
--- a/fdbsyncd/fdbsync.h
+++ b/fdbsyncd/fdbsync.h
@@ -26,9 +26,9 @@ enum FDB_TYPE {
 
 struct m_fdb_info
 {
-    char mac[32];
+    std::string  mac;
     std::string  vid;           /*Store as Vlan<ID> */
-    char port_name[32];
+    std::string  port_name;
     short type;                 /*dynamic or static*/
     short op_type;              /*add or del*/
 };
